gbuiltins/staking/src/lib.rs

```diff
@@ -0,0 +1,128 @@
+//! ));
+//!
+//! #[gstd::async_main]
+//! async fn main() {
+//!     let value = msg::value();
+//!     let payee: Option<RewardAccount> = None;
+//!     let payload = Request::V1(RequestV1::Bond { value, payee }).encode();
+//!     let _ = msg::send_bytes_for_reply(BUILTIN_ADDRESS, &payload[..], 0, 0)
+//!         .expect("Error sending message")
+//!         .await;
+//! }
+//! # fn main() {}
+//! ```
+//!
+
+#![no_std]
+
+extern crate alloc;
+
+use alloc::vec::Vec;
+use scale_info::scale::{self, Decode, Encode};
+
+pub type AccountId = [u8; 32];
+
+/// Type that should be used to create a message to the staking built-in actor.
+///
+/// A [partial] mirror of the staking pallet interface. Not all extrinsics
+/// are supported, more can be added as needed for real-world use cases.
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug)]
+#[codec(crate = scale)]
+pub enum Request {
+    /// Version 1 of the staking built-in actor protocol.
+    V1(RequestV1),
+}
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug)]
+#[codec(crate = scale)]
+pub enum RequestV1 {
+    /// Bond up to the `value` from the sender to self as the controller.
+    Bond {
+        value: u128,
+        payee: Option<RewardAccount>,
+    },
+    /// Add up to the `value` to the sender's bonded amount.
+    BondExtra { value: u128 },
+    /// Unbond up to the `value` to allow withdrawal after undonding period.
+    Unbond { value: u128 },
+    /// Withdraw unbonded chunks for which undonding period has elapsed.
+    WithdrawUnbonded { num_slashing_spans: u32 },
+    /// Add sender as a nominator of `targets` or update the existing targets.
+    Nominate { targets: Vec<AccountId> },
+    /// Declare intention to [temporarily] stop nominating while still having funds bonded.
+    Chill,
+    /// Request stakers payout for the given era.
+    PayoutStakers {
+        validator_stash: AccountId,
+        era: u32,
+    },
+    /// Rebond a portion of the sender's stash scheduled to be unlocked.
+    Rebond { value: u128 },
+    /// Set the reward destination.
+    SetPayee { payee: RewardAccount },
+}
+
+/// An account where the rewards should accumulate on.
+///
+/// In order to separate the contract's own balance from the rewards earned by users funds,
+/// a separate account for the rewards can be assigned.
+#[derive(Encode, Decode, Clone, Copy, PartialEq, Eq, Debug)]
+#[codec(crate = scale)]
+pub enum RewardAccount {
+    /// Accumulate the rewards on contract's account derived from its `program_id`.
+    Program,
+    /// Accumulate the rewards on a separate account.
+    Custom(AccountId),
+}
+
+/// Message processing output
+pub type Response = ();
+
+/// Type representing a dispatched Runtime call internal error.
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, derive_more::Display)]
+#[codec(crate = scale)]
+pub enum DispatchErrorReason {
+    #[display(fmt = "Runtime internal error")]
+    RuntimeError,
+}
```

Comment:
these two types looks useless for me too

Reply:
Just abstractions. With zero runtime overhead. These particular ones might seem trivial but they could be more complex.